File: C2/PDOO/Practicas/DeepSpaceJava/src/deepspace/EnemyStarShip.java

package deepspace;

/**
 *
 * @author Pedro Pablo Ruiz Huertas y Juan Antonio Villegas Recio
 */
public class EnemyStarShip implements Copyable <EnemyStarShip>, SpaceFighter {
    private float ammoPower;
    private String name;
    private float shieldPower;
    private Damage damage;
    private Loot loot;
    
    EnemyStarShip(String n, float a, float s, Loot l, Damage d){
        this.ammoPower=a;
        this.name=n;
        this.shieldPower=s;
        this.loot=l;
        this.damage=d;
    }
    
    EnemyStarShip(EnemyStarShip s){
        this.ammoPower=s.ammoPower;
        this.name=s.name;
        this.shieldPower=s.shieldPower;
        this.loot=s.loot;
        this.damage=s.damage;
    }
    
    public String getName() {
        return this.name;
    }

    public float getAmmoPower() {
        return this.ammoPower;
    }

    public float getShieldPower() {
        return this.shieldPower;
    }

    public Loot getLoot() {
        return this.loot;
    }

    public Damage getDamage() {
        return this.damage;
    }
    
    
    public EnemyToUI getUIversion(){
        return new EnemyToUI(this);
    }
    
    @Override
    public float fire(){
        return this.ammoPower;
    }
    
    @Override
    public float protection(){
        return this.shieldPower;
    }
    
    @Override
    public ShotResult receiveShot(float shot){
        if(this.shieldPower<shot)
            return ShotResult.DONOTRESIST;
        else return ShotResult.RESIST;
    }
    
    @Override
    public String toString(){
        String mensaje="El nombre de esta nave enemiga es:"+name+
                        ",\nel poder del arma de esa nave enemiga es:"+ammoPower+
                        ",\nel poder del escudo es: "+shieldPower+
                        ",\nel botin que contiene es:"+loot.toString()+
                        ",\ny el daño es:"+damage.toString();
        return mensaje;
    }
    
    @Override
    public EnemyStarShip copy(){
        return new EnemyStarShip(this);
    }
}

File: C2/PDOO/Practicas/DeepSpaceRuby/lib/EnemyStarShip.rb
#encoding:utf-8


require_relative "EnemyToUI"
require_relative "Damage"
require_relative "Loot"
require_relative "ShotResult"

module Deepspace
  class EnemyStarShip
    
    attr_reader :ammoPower, :name, :shieldPower, :loot, :damage
    public
    
    def initialize(n,a,s,l,d)
      @name=n
      @ammoPower=a
      @shieldPower=s
      @loot=l
      @damage=d
    end
    
    def self.newCopy(e)
      new(e.name,e.ammoPower,e.shieldPower,e.loot,e.damage)
    end
    
    def getUIversion
      return EnemyToUI.new(self)
    end
    
    def fire
      return @ammoPower
    end
    
    def protection
      return @shieldPower
    end
    
    def receiveShot(shot)
      if @shieldPower<shot
        return ShotResult::DONOTRESIST
      else
        return ShotResult::RESIST
      end
    end
    
    def to_s
      return "\nNombre: #{@name}, \nDano: "+@damage.to_s+"\nBotin: "+@loot.to_s+"\nEnergia del escudo: #{@shieldPower} \nPotencia de disparo: #{@ammoPower}\n\n"
    end
    
  end
end

